### PR TITLE
Use actual directive def'n instead of mock.

### PR DESCRIPTION
**Pull request by kegluneq**
Per earlier request & TODO in angular2/test/transform/common.dart

### CODE REVIEW

**Comment by mhevery** [comment]
Please format your SHAs per https://github.com/angular/angular.js/blob/master/CONTRIBUTING.md#commit-message-format

File: modules/angular2/test/transform/ctor_with_default_value_files/bar.dart [hunk 1]
```diff
@@ -2,8 +2,8 @@ library bar;
 
 import 'package:angular2/src/core/annotations/annotations.dart';
 
-@Directive(context: 'soup')
-class Component {
+@Component(selector: '[soup]')
+class MyComponent {
   final dynamic c;
-  Component([this.c = 'sandwich']);
+  MyComponent([this.c = 'sandwich']);
```
Comment (mhevery) [inline modules/angular2/test/transform/ctor_with_default_value_files/bar.dart]: An actual component will never have variable length parameters, as it is not compatible with DI.

File: modules/angular2/test/transform/ctor_with_default_value_files/expected/index.bootstrap.dart [hunk 2]
```diff
@@ -5,10 +5,10 @@ import 'index.dart' as i2;
 
 main() {
   reflector
-    ..registerType(i0.Component, {
-      "factory": (dynamic c) => new i0.Component(c),
+    ..registerType(i0.MyComponent, {
+      "factory": (dynamic c) => new i0.MyComponent(c),
```
Comment (mhevery) [inline modules/angular2/test/transform/ctor_with_default_value_files/expected/index.bootstrap.dart]: Could we give `c` property some type information and than show that the generated stub as the same type?

File: modules/angular2/test/transform/list_of_types_files/bar.dart [hunk 3]
```diff
@@ -3,8 +3,8 @@ library bar;
 import 'package:angular2/src/core/annotations/annotations.dart';
 import 'foo.dart';
 
-@Directive(context: const [MyContext])
-class Component {
+@Component(lightDomServices: const [MyContext])
```
Comment (mhevery) [inline modules/angular2/test/transform/list_of_types_files/bar.dart]: `lightDomServices` are going away, use `componentServices` instead.

File: modules/angular2/test/transform/two_deps_files/bar.dart [hunk 4]
```diff
@@ -3,11 +3,11 @@ library bar;
 import 'package:angular2/src/core/annotations/annotations.dart';
 import 'foo.dart';
 
-@Directive(context: const MyContext(contextString))
-class Component2 {
+@Component(selector: const MyContext(contextString))
```
Comment (mhevery) [inline modules/angular2/test/transform/two_deps_files/bar.dart]: `selector` will always be a string. Not sure what `MyContext` represents here. This should be a analyzer error but we have to switch this types: https://github.com/angular/angular/blob/master/modules/angular2/src/core/annotations/annotations.js#L6

**Comment by kegluneq** [comment]
I believe I've addressed all comments - PTAL.

**Pull request closed without merging**